1_fetch/src/get_sb_data.R

```diff
@@ -206,18 +72,95 @@ download_children <-function(sites, sb_table_reduced, dldir, workdir, outdir, ou
   
   itemfails <- as.data.frame(itemfails)
   
-  filepath1 <- file.path(outdir, out_file_name)
+  filepath1 <- paste0(outdir, "/", out_file_name, item, ".csv")
   write_csv(data_at_sites, filepath1)
-  filepath2 <- file.path(outdir, paste0("FAILS_", out_file_name))
+  filepath2 <- paste0(outdir, "/", out_file_name, item, "_FAILS.csv")
   write_csv(itemfails, filepath2)
   return_df = c(filepath1, filepath2)
   return(return_df)
 }
 
-calc_avg_dams <- function(sb_data) {
-  sb_data <- read_csv(sb_data, show_col_types = FALSE) %>%
-    suppressMessages()
-  dams <- sb_data %>%
+get_sb_data_log_g2 <- function(sb_var_ids, file_out) {
+  
+  sb_log <- tibble(sb_id = character(), 
+                   last_update = character())
+  for (i in 1:nrow(sb_var_ids)) {
+    sb_id <- as.character(sb_var_ids[i,1])
+    last_update <- item_get_fields(sb_id, "provenance")$lastUpdated
+    log_item <- tibble(sb_id, last_update)
+    sb_log <- bind_rows(sb_log, log_item)
+  }
+  
+  write_csv(sb_log, file_out)
+  return(file_out)
+  
+}
+
+prep_feature_vars <- function(sb_var_data, sites) {
+  
+  data <- sites %>%
+    select(COMID, LAT, LON, npdes, fwwd, strg, devl, cndp)
+  
+  for (i in 1:length(sb_var_data)) {
+    filepath <- sb_var_data[[i]][1]
+    data_temp <- read_csv(filepath, show_col_types = FALSE) %>%
+      group_by(COMID) %>%
+      summarise_all(mean)
+    data <- left_join(data, data_temp, by = "COMID")
+  }
+  
+  data$npdes <- as.numeric(data$npdes)
+  data$fwwd <- as.numeric(data$fwwd)
+  data$strg <- as.numeric(data$strg)
+  data$devl <- as.numeric(data$devl)
+  data$cndp <- as.numeric(data$cndp)
+  
+  data <- data %>% 
+    group_by(COMID) %>%
+    summarise_all(mean)
```

Comment:
Is this because there are duplicate COMIDs? Can you add a comment?

Reply:
Correct. All the feature variable values should be the same (I skimmed, but didn't formally check - that would be easy to do). So here, mean of all simply eliminates the duplicates. I will verify and comment.